src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBEmulatorHealthCheck.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Aspire.Hosting.Azure.CosmosDB;
+
+/// <summary>
+/// This health check also creates default databases and containers for the Azure CosmosDB Emulator.
```

Comment:
This feels like too much responsibility. Everytime we are checking for "health" we are rechecking all the dbs and containers.

I think it would be better if the health check was only a health check. And then we "waited" for it to be healthy and only ensure the db and containers are created once.

Reply:
The default health check is checking only one db and its containers (because its state is only for one db, I assume one could create more instances for other dbs). At least we could do the same, pick the first db.

But I did what you suggested at first, though this could introduce undesired effects that it's stated as healthy but we haven't created the db/containers yet. So while we are creating them other services could have started using them.

Another option would be to change the implementation to do like it does right now, but then store a bool and only check for the first db afterwards (not even the containers).

Reply:
Added a flag to only check/create for the resources once.

Reply:
This seems like it's the only point of contention in the PR. Is it resolved? The other changes look good.

Reply:
We should have a live chat about it, I tried different options. Right now it does the creation attempt only once then falls back to the standard check. This is just the apphost health check though to report for `WaithHealthyAsync`, the clients use the standard health check.

Reply:
I approved the PR, can you open an issue? It's kinda related to  https://github.com/dotnet/aspire/issues/7101